### ISSUE TITLE
rspec command failing in my Rails 5 app after upgrading Ruby 2.3.2 to 2.4.0 

### ISSUE THREAD

**Issue by jiggneshhgohel** [rspec/rspec-dev#176]
**.ruby-version**

```
2.4.0
```

**.ruby-gemset**

```
my_app-rails-5
```

**Gemfile**

```
source 'https://rubygems.org'


gem 'rails', '~> 5.0', '>= 5.0.0.1'

gem 'pg'

# Use SCSS for stylesheets
gem 'sass-rails', '~> 5.0'
gem 'bootstrap-sass', '~> 3.3.1'
gem 'autoprefixer-rails'

# Use Uglifier as compressor for JavaScript assets
gem 'uglifier', '>= 1.3.0'
# Use CoffeeScript for .coffee assets and views
gem 'coffee-rails', '~> 4.1.0'
# See https://github.com/sstephenson/execjs#readme for more supported runtimes
gem 'therubyracer', platforms: :ruby

# Use jquery as the JavaScript library
gem 'jquery-rails'
# Turbolinks makes following links in your web application faster. Read more: https://github.com/rails/turbolinks
gem 'turbolinks'

# bundle exec rake doc:rails generates the API under doc/api.
gem 'sdoc', '~> 0.4.0', group: :doc

gem "haml-rails", "~> 0.9"

gem 'devise'
gem 'simple_token_authentication'
gem 'jwt'
gem 'versionist'
gem 'twilio-ruby', '~> 4.11.1'
gem 'config'
gem 'rabl-rails'
gem 'rest-client'
gem 'pundit'
gem 'julia_builder'
gem 'rack-cors', :require => 'rack/cors'
# This is Side-Kiq
gem 'sidekiq'
# This is Side-Tiq
gem 'sidetiq'
# if you require 'sinatra' you get the DSL extended to Object
gem 'sinatra', :require => false

gem "airbrussh", :require => false

gem 'hirb', '~> 0.7.3'
gem 'json', '1.8.6'

group :development do
  gem 'capistrano'
  gem 'capistrano-bundler'
  gem 'capistrano-rails'
  gem 'capistrano-rbenv', github: 'capistrano/rbenv'

  # Access an IRB console on exception pages or by using <%= console %> in views
  gem 'web-console', '~> 2.0'
end

group :development, :test, :staging, :production, :staging_experimental do
  gem 'database_cleaner'
end

group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  gem 'byebug'

  gem 'rspec-rails', '~> 3.1'
  gem 'shoulda-matchers', '~> 3.0'
  gem 'factory_girl_rails', "~> 4.0"
  gem 'ffaker'
  gem 'rails-controller-testing'

  gem 'json_expressions'
end

group :test do
  gem 'webmock'
end
```

**Gemfile.lock**

```
GIT
  remote: git://github.com/capistrano/rbenv.git
  revision: e056efc0c361cc15b76ff961d6c5f65ed5e2c57e
  specs:
    capistrano-rbenv (2.0.4)
      capistrano (~> 3.1)
      sshkit (~> 1.3)

GEM
  remote: https://rubygems.org/
  specs:
    actioncable (5.0.0.1)
      actionpack (= 5.0.0.1)
      nio4r (~> 1.2)
      websocket-driver (~> 0.6.1)
    actionmailer (5.0.0.1)
      actionpack (= 5.0.0.1)
      actionview (= 5.0.0.1)
      activejob (= 5.0.0.1)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (5.0.0.1)
      actionview (= 5.0.0.1)
      activesupport (= 5.0.0.1)
      rack (~> 2.0)
      rack-test (~> 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    actionview (5.0.0.1)
      activesupport (= 5.0.0.1)
      builder (~> 3.1)
      erubis (~> 2.7.0)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    activejob (5.0.0.1)
      activesupport (= 5.0.0.1)
      globalid (>= 0.3.6)
    activemodel (5.0.0.1)
      activesupport (= 5.0.0.1)
    activerecord (5.0.0.1)
      activemodel (= 5.0.0.1)
      activesupport (= 5.0.0.1)
      arel (~> 7.0)
    activesupport (5.0.0.1)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (~> 0.7)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
    addressable (2.4.0)
    airbrussh (1.1.1)
      sshkit (>= 1.6.1, != 1.7.0)
    arel (7.1.2)
    autoprefixer-rails (6.5.0.2)
      execjs
    bcrypt (3.1.11)
    binding_of_caller (0.7.2)
      debug_inspector (>= 0.0.1)
    bootstrap-sass (3.3.7)
      autoprefixer-rails (>= 5.2.1)
      sass (>= 3.3.4)
    builder (3.2.2)
    byebug (9.0.6)
    capistrano (3.6.1)
      airbrussh (>= 1.0.0)
      capistrano-harrow
      i18n
      rake (>= 10.0.0)
      sshkit (>= 1.9.0)
    capistrano-bundler (1.2.0)
      capistrano (~> 3.1)
      sshkit (~> 1.2)
    capistrano-harrow (0.5.3)
    capistrano-rails (1.1.8)
      capistrano (~> 3.1)
      capistrano-bundler (~> 1.1)
    celluloid (0.17.3)
      celluloid-essentials
      celluloid-extras
      celluloid-fsm
      celluloid-pool
      celluloid-supervision
      timers (>= 4.1.1)
    celluloid-essentials (0.20.5)
      timers (>= 4.1.1)
    celluloid-extras (0.20.5)
      timers (>= 4.1.1)
    celluloid-fsm (0.20.5)
      timers (>= 4.1.1)
    celluloid-pool (0.20.5)
      timers (>= 4.1.1)
    celluloid-supervision (0.20.6)
      timers (>= 4.1.1)
    coffee-rails (4.1.1)
      coffee-script (>= 2.2.0)
      railties (>= 4.0.0, < 5.1.x)
    coffee-script (2.4.1)
      coffee-script-source
      execjs
    coffee-script-source (1.10.0)
    concurrent-ruby (1.0.2)
    config (1.3.0)
      activesupport (>= 3.0)
      deep_merge (~> 1.1.1)
    connection_pool (2.2.0)
    crack (0.4.3)
      safe_yaml (~> 1.0.0)
    database_cleaner (1.5.3)
    debug_inspector (0.0.2)
    deep_merge (1.1.1)
    devise (4.2.0)
      bcrypt (~> 3.0)
      orm_adapter (~> 0.1)
      railties (>= 4.1.0, < 5.1)
      responders
      warden (~> 1.2.3)
    diff-lcs (1.2.5)
    domain_name (0.5.20160826)
      unf (>= 0.0.5, < 1.0.0)
    erubis (2.7.0)
    execjs (2.7.0)
    factory_girl (4.7.0)
      activesupport (>= 3.0.0)
    factory_girl_rails (4.7.0)
      factory_girl (~> 4.7.0)
      railties (>= 3.0.0)
    ffaker (2.2.0)
    globalid (0.3.7)
      activesupport (>= 4.1.0)
    haml (4.0.7)
      tilt
    haml-rails (0.9.0)
      actionpack (>= 4.0.1)
      activesupport (>= 4.0.1)
      haml (>= 4.0.6, < 5.0)
      html2haml (>= 1.0.1)
      railties (>= 4.0.1)
    hashdiff (0.3.0)
    hirb (0.7.3)
    hitimes (1.2.4)
    html2haml (2.0.0)
      erubis (~> 2.7.0)
      haml (~> 4.0.0)
      nokogiri (~> 1.6.0)
      ruby_parser (~> 3.5)
    http-cookie (1.0.3)
      domain_name (~> 0.5)
    i18n (0.7.0)
    ice_cube (0.14.0)
    jquery-rails (4.2.1)
      rails-dom-testing (>= 1, < 3)
      railties (>= 4.2.0)
      thor (>= 0.14, < 2.0)
    json (1.8.6)
    json_expressions (0.8.3)
    julia_builder (0.2.0)
    jwt (1.5.6)
    libv8 (3.16.14.15)
    loofah (2.0.3)
      nokogiri (>= 1.5.9)
    mail (2.6.4)
      mime-types (>= 1.16, < 4)
    method_source (0.8.2)
    mime-types (3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2016.0521)
    mini_portile2 (2.1.0)
    minitest (5.9.1)
    multi_json (1.12.1)
    net-scp (1.2.1)
      net-ssh (>= 2.6.5)
    net-ssh (3.2.0)
    netrc (0.11.0)
    nio4r (1.2.1)
    nokogiri (1.6.8.1)
      mini_portile2 (~> 2.1.0)
    orm_adapter (0.5.0)
    pg (0.19.0)
    pundit (1.1.0)
      activesupport (>= 3.0.0)
    rabl-rails (0.4.3)
      activesupport (>= 3.1)
      railties (>= 3.1)
      thread_safe (~> 0.3.1)
    rack (2.0.1)
    rack-cors (0.4.0)
    rack-protection (1.5.3)
      rack
    rack-test (0.6.3)
      rack (>= 1.0)
    rails (5.0.0.1)
      actioncable (= 5.0.0.1)
      actionmailer (= 5.0.0.1)
      actionpack (= 5.0.0.1)
      actionview (= 5.0.0.1)
      activejob (= 5.0.0.1)
      activemodel (= 5.0.0.1)
      activerecord (= 5.0.0.1)
      activesupport (= 5.0.0.1)
      bundler (>= 1.3.0, < 2.0)
      railties (= 5.0.0.1)
      sprockets-rails (>= 2.0.0)
    rails-controller-testing (1.0.1)
      actionpack (~> 5.x)
      actionview (~> 5.x)
      activesupport (~> 5.x)
    rails-dom-testing (2.0.1)
      activesupport (>= 4.2.0, < 6.0)
      nokogiri (~> 1.6.0)
    rails-html-sanitizer (1.0.3)
      loofah (~> 2.0)
    railties (5.0.0.1)
      actionpack (= 5.0.0.1)
      activesupport (= 5.0.0.1)
      method_source
      rake (>= 0.8.7)
      thor (>= 0.18.1, < 2.0)
    rake (11.3.0)
    rdoc (4.2.2)
      json (~> 1.4)
    redis (3.3.1)
    ref (2.0.0)
    responders (2.3.0)
      railties (>= 4.2.0, < 5.1)
    rest-client (2.0.0)
      http-cookie (>= 1.0.2, < 2.0)
      mime-types (>= 1.16, < 4.0)
      netrc (~> 0.8)
    rspec-core (3.5.4)
      rspec-support (~> 3.5.0)
    rspec-expectations (3.5.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.5.0)
    rspec-mocks (3.5.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.5.0)
    rspec-rails (3.5.2)
      actionpack (>= 3.0)
      activesupport (>= 3.0)
      railties (>= 3.0)
      rspec-core (~> 3.5.0)
      rspec-expectations (~> 3.5.0)
      rspec-mocks (~> 3.5.0)
      rspec-support (~> 3.5.0)
    rspec-support (3.5.0)
    ruby_parser (3.8.2)
      sexp_processor (~> 4.1)
    safe_yaml (1.0.4)
    sass (3.4.22)
    sass-rails (5.0.6)
      railties (>= 4.0.0, < 6)
      sass (~> 3.1)
      sprockets (>= 2.8, < 4.0)
      sprockets-rails (>= 2.0, < 4.0)
      tilt (>= 1.1, < 3)
    sdoc (0.4.1)
      json (~> 1.7, >= 1.7.7)
      rdoc (~> 4.0)
    sexp_processor (4.7.0)
    shoulda-matchers (3.1.1)
      activesupport (>= 4.0.0)
    sidekiq (4.2.2)
      concurrent-ruby (~> 1.0)
      connection_pool (~> 2.2, >= 2.2.0)
      rack-protection (~> 1.5)
      redis (~> 3.2, >= 3.2.1)
    sidetiq (0.7.2)
      celluloid (>= 0.17.3)
      ice_cube (~> 0.14.0)
      sidekiq (>= 4.1.0)
    simple_token_authentication (1.14.0)
      actionmailer (>= 3.2.6, < 6)
      actionpack (>= 3.2.6, < 6)
      devise (>= 3.2, < 6)
    sinatra (1.0)
      rack (>= 1.0)
    sprockets (3.7.0)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.2.0)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    sshkit (1.11.3)
      net-scp (>= 1.1.2)
      net-ssh (>= 2.8.0)
    therubyracer (0.12.2)
      libv8 (~> 3.16.14.0)
      ref
    thor (0.19.1)
    thread_safe (0.3.5)
    tilt (2.0.5)
    timers (4.1.1)
      hitimes
    turbolinks (5.0.1)
      turbolinks-source (~> 5)
    turbolinks-source (5.0.0)
    twilio-ruby (4.11.1)
      builder (>= 2.1.2)
      jwt (~> 1.0)
      multi_json (>= 1.3.0)
    tzinfo (1.2.2)
      thread_safe (~> 0.1)
    uglifier (3.0.2)
      execjs (>= 0.3.0, < 3)
    unf (0.1.4)
      unf_ext
    unf_ext (0.0.7.2)
    versionist (1.5.0)
      activesupport (>= 3)
      railties (>= 3)
      yard (~> 0.7)
    warden (1.2.6)
      rack (>= 1.0)
    web-console (2.3.0)
      activemodel (>= 4.0)
      binding_of_caller (>= 0.7.2)
      railties (>= 4.0)
      sprockets-rails (>= 2.0, < 4.0)
    webmock (2.1.0)
      addressable (>= 2.3.6)
      crack (>= 0.3.2)
      hashdiff
    websocket-driver (0.6.4)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.2)
    yard (0.9.5)

PLATFORMS
  ruby

DEPENDENCIES
  airbrussh
  autoprefixer-rails
  bootstrap-sass (~> 3.3.1)
  byebug
  capistrano
  capistrano-bundler
  capistrano-rails
  capistrano-rbenv!
  coffee-rails (~> 4.1.0)
  config
  database_cleaner
  devise
  factory_girl_rails (~> 4.0)
  ffaker
  haml-rails (~> 0.9)
  hirb (~> 0.7.3)
  jquery-rails
  json (= 1.8.6)
  json_expressions
  julia_builder
  jwt
  pg
  pundit
  rabl-rails
  rack-cors
  rails (~> 5.0, >= 5.0.0.1)
  rails-controller-testing
  rest-client
  rspec-rails (~> 3.1)
  sass-rails (~> 5.0)
  sdoc (~> 0.4.0)
  shoulda-matchers (~> 3.0)
  sidekiq
  sidetiq
  simple_token_authentication
  sinatra
  therubyracer
  turbolinks
  twilio-ruby (~> 4.11.1)
  uglifier (>= 1.3.0)
  versionist
  web-console (~> 2.0)
  webmock

BUNDLED WITH
   1.14.4

```

Trying to run `rspec` from the root of my Rails 5 application I am getting following error:

```
my_app$ rspec
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/concurrent-ruby-1.0.2/lib/concurrent/map.rb:206: warning: constant ::Fixnum is deprecated
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/activesupport-5.0.0.1/lib/active_support/xml_mini.rb:51: warning: constant ::Fixnum is deprecated
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/activesupport-5.0.0.1/lib/active_support/xml_mini.rb:52: warning: constant ::Bignum is deprecated
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/sprockets-3.7.0/lib/sprockets/digest_utils.rb:47: warning: constant ::Fixnum is deprecated
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/sprockets-3.7.0/lib/sprockets/digest_utils.rb:51: warning: constant ::Bignum is deprecated
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/sprockets-3.7.0/lib/sprockets/processor_utils.rb:110: warning: constant ::Fixnum is deprecated
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/sprockets-3.7.0/lib/sprockets/processor_utils.rb:111: warning: constant ::Bignum is deprecated
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/therubyracer-0.12.2/lib/v8/conversion.rb:21: warning: constant ::Fixnum is deprecated
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler/runtime.rb:94:in `rescue in block (2 levels) in require': There was an error while trying to load the gem 'autoprefixer-rails'. (Bundler::GemRequireError)
Gem Load Error is: wrong argument type Class (expected Module)
Backtrace for gem load error is:
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/therubyracer-0.12.2/lib/v8/conversion.rb:23:in `include'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/therubyracer-0.12.2/lib/v8/conversion.rb:23:in `block (2 levels) in <top (required)>'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/therubyracer-0.12.2/lib/v8/conversion.rb:22:in `class_eval'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/therubyracer-0.12.2/lib/v8/conversion.rb:22:in `block in <top (required)>'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/therubyracer-0.12.2/lib/v8/conversion.rb:21:in `each'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/therubyracer-0.12.2/lib/v8/conversion.rb:21:in `<top (required)>'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/therubyracer-0.12.2/lib/v8.rb:22:in `require'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/therubyracer-0.12.2/lib/v8.rb:22:in `<top (required)>'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/execjs-2.7.0/lib/execjs/ruby_racer_runtime.rb:108:in `require'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/execjs-2.7.0/lib/execjs/ruby_racer_runtime.rb:108:in `available?'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/execjs-2.7.0/lib/execjs/runtimes.rb:63:in `each'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/execjs-2.7.0/lib/execjs/runtimes.rb:63:in `find'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/execjs-2.7.0/lib/execjs/runtimes.rb:63:in `best_available'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/execjs-2.7.0/lib/execjs/runtimes.rb:57:in `autodetect'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/execjs-2.7.0/lib/execjs.rb:5:in `<module:ExecJS>'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/execjs-2.7.0/lib/execjs.rb:4:in `<top (required)>'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/autoprefixer-rails-6.5.0.2/lib/autoprefixer-rails/processor.rb:2:in `require'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/autoprefixer-rails-6.5.0.2/lib/autoprefixer-rails/processor.rb:2:in `<top (required)>'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/autoprefixer-rails-6.5.0.2/lib/autoprefixer-rails.rb:35:in `require_relative'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/autoprefixer-rails-6.5.0.2/lib/autoprefixer-rails.rb:35:in `<top (required)>'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler/runtime.rb:91:in `require'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler/runtime.rb:91:in `block (2 levels) in require'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler/runtime.rb:86:in `each'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler/runtime.rb:86:in `block in require'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler/runtime.rb:75:in `each'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler/runtime.rb:75:in `require'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler.rb:107:in `require'
/home/jignesh/ror_apps/my_app/config/application.rb:7:in `<top (required)>'
/home/jignesh/ror_apps/my_app/config/environment.rb:2:in `require_relative'
/home/jignesh/ror_apps/my_app/config/environment.rb:2:in `<top (required)>'
/home/jignesh/ror_apps/my_app/spec/rails_helper.rb:4:in `require'
/home/jignesh/ror_apps/my_app/spec/rails_helper.rb:4:in `<top (required)>'
/home/jignesh/ror_apps/my_app/spec/controllers/api/authentication/sessions_controller_spec.rb:1:in `require'
/home/jignesh/ror_apps/my_app/spec/controllers/api/authentication/sessions_controller_spec.rb:1:in `<top (required)>'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/configuration.rb:1435:in `load'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/configuration.rb:1435:in `block in load_spec_files'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/configuration.rb:1433:in `each'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/configuration.rb:1433:in `load_spec_files'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:100:in `setup'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:86:in `run'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:71:in `run'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:45:in `invoke'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/exe/rspec:4:in `<top (required)>'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/bin/rspec:22:in `load'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/bin/rspec:22:in `<main>'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/bin/ruby_executable_hooks:15:in `eval'
/home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/bin/ruby_executable_hooks:15:in `<main>'
Bundler Error Backtrace:
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler/runtime.rb:90:in `block (2 levels) in require'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler/runtime.rb:86:in `each'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler/runtime.rb:86:in `block in require'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler/runtime.rb:75:in `each'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler/runtime.rb:75:in `require'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/bundler-1.14.4/lib/bundler.rb:107:in `require'
  from /home/jignesh/ror_apps/my_app/config/application.rb:7:in `<top (required)>'
  from /home/jignesh/ror_apps/my_app/config/environment.rb:2:in `require_relative'
  from /home/jignesh/ror_apps/my_app/config/environment.rb:2:in `<top (required)>'
  from /home/jignesh/ror_apps/my_app/spec/rails_helper.rb:4:in `require'
  from /home/jignesh/ror_apps/my_app/spec/rails_helper.rb:4:in `<top (required)>'
  from /home/jignesh/ror_apps/my_app/spec/controllers/api/authentication/sessions_controller_spec.rb:1:in `require'
  from /home/jignesh/ror_apps/my_app/spec/controllers/api/authentication/sessions_controller_spec.rb:1:in `<top (required)>'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/configuration.rb:1435:in `load'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/configuration.rb:1435:in `block in load_spec_files'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/configuration.rb:1433:in `each'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/configuration.rb:1433:in `load_spec_files'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:100:in `setup'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:86:in `run'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:71:in `run'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:45:in `invoke'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/rspec-core-3.5.4/exe/rspec:4:in `<top (required)>'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/bin/rspec:22:in `load'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/bin/rspec:22:in `<main>'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/bin/ruby_executable_hooks:15:in `eval'
  from /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/bin/ruby_executable_hooks:15:in `<main>'
```

Is this a known issue?

**Note**

 While upgrading from Ruby 2.3.2 to 2.4.0 and running `bundle` I had to explicitly use `gem 'json', '1.8.6'` because of following error:

```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/json-1.8.3/ext/json/ext/generator
/home/jignesh/.rvm/rubies/ruby-2.4.0/bin/ruby -r ./siteconf20170214-22510-1hekbu3.rb extconf.rb 
creating Makefile

current directory: /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/json-1.8.3/ext/json/ext/generator
make "DESTDIR=" clean

current directory: /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/json-1.8.3/ext/json/ext/generator
make "DESTDIR="
compiling generator.c
generator.c: In function ‘generate_json’:
generator.c:861:25: error: ‘rb_cFixnum’ undeclared (first use in this function)
     } else if (klass == rb_cFixnum) {
                         ^
generator.c:861:25: note: each undeclared identifier is reported only once for each function it appears in
generator.c:863:25: error: ‘rb_cBignum’ undeclared (first use in this function)
     } else if (klass == rb_cBignum) {
                         ^
generator.c: At top level:
cc1: warning: unrecognized command line option "-Wno-self-assign" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-constant-logical-operand" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-parentheses-equality" [enabled by default]
cc1: warning: unrecognized command line option "-Wno-tautological-compare" [enabled by default]
make: *** [generator.o] Error 1

make failed, exit code 2

Gem files will remain installed in /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/gems/json-1.8.3 for inspection.
Results logged to /home/jignesh/.rvm/gems/ruby-2.4.0@my_app-rails-5/extensions/x86_64-linux/2.4.0/json-1.8.3/gem_make.out

An error occurred while installing json (1.8.3), and Bundler cannot continue.
Make sure that `gem install json -v '1.8.3'` succeeds before bundling.
```

searching for which I ended up at https://github.com/flori/json/issues/303 and while experimenting with suggested solutions finally using `-v 1.8.3` of the `json` gem helped the `bundle` command execution to be successful

**Comment by jiggneshhgohel** [comment]
I just noticed that the same error occurs when my CI tool tries to run `bundle exec rake db:schema:load` as part of automated deployment. Pardon me if this is not the right place for this issue. And please if anybody has any idea regarding to which repository this issue belongs to then please let me know and I can post this there.

Thanks.

**Comment by myronmarston** [comment]
The error you're reporting is in therubyracer (that's where an `include` statement is attempting to include a class instead of a module).  It was reported and fixed in cowboyd/therubyracer#413, and that fix is included in therubyracer v0.12.3, so if you upgrade that gem, it should fix the issue.

In the future, please dig a bit more into the issue (and try running your code outside of RSpec) to see if RSpec has anything to do with it before reporting an issue here.

**Comment by jiggneshhgohel** [comment]
Thanks @myronmarston. I will try your suggestion and yes definitely next time I will dig more before posting the issue.
